### PR TITLE
on_entry_dbg_property_pattern_activate is not actually called

### PR DESCRIPTION
**Pull request by marcodiegomesquita**
Fix by conveniently calling it when needed.

### CODE REVIEW

**Comment by rmanfredi** [comment]
This pull request contains fixes already merged, and this creates conflicts.
You should remove the first 3 offending commits and keep only the last one.

File: src/ui/gtk/settings_cb.c [hunk 1]
```diff
@@ -951,7 +959,7 @@ void
 on_entry_dbg_property_pattern_activate(GtkEditable *unused_editable,
 	gpointer unused_udata)
 {
-	static gchar old_pattern[1024];
+	static gchar old_pattern[1024] = {0};
```
Comment (rmanfredi) [inline src/ui/gtk/settings_cb.c:962]: Can you put spaces around braces and say:
`{ '\0' }`
to make it more obvious that we are initializing an array of characters

File: src/ui/gtk/settings_cb.c [hunk 2]
```diff
@@ -962,7 +970,7 @@ on_entry_dbg_property_pattern_activate(GtkEditable *unused_editable,
         0, -1));
 	g_strstrip(text);
 
-	if (0 != strcmp(text, old_pattern)) {
+	if (0 != strcmp(text, old_pattern) || text[0] == '\0') {
```
Comment (rmanfredi) [inline src/ui/gtk/settings_cb.c:973]: It is best to say
`'\0' == text[0]`
That way, if you make a typo and forget one '=' you get an invalid assignment and therefore an error.
Plus it puts emphasis on the value being tested.

**Comment by marcodiegomesquita** [comment]
Updated with requested changes.

**Comment by rmanfredi** [comment]
Great, merging!
Thanks again for your contribution.

**Pull request merged**